frontend/src/lib/docs.ts

```diff
@@ -6,19 +6,14 @@ import Swagger from '@apidevtools/swagger-parser';
 import { OpenAPIV2 } from 'openapi-types';
 import { request } from './k8s/apiProxy';
 
-let docsPromise: ReturnType<typeof getDocs>;
-
-async function getDocs() {
+export async function getDocs() {
   const docs = await request('/openapi/v2');
   return Swagger.dereference(docs);
 }
 
-export default async function getDocDefinitions(apiVersion: string, kind: string) {
-  if (!docsPromise) {
-    docsPromise = getDocs(); // Don't wait here. Just kick off the request
-  }
-
-  const { definitions = {} } = (await docsPromise) as OpenAPIV2.Document;
+export async function getDocDefinitions(apiVersion: string, kind: string) {
+  // We want a fresh call to getDocs() each time to ensure we get the latest mock data during testing
```

Comment:
this changes the behavior of the 'getDocDefinitions' function, have you tested that the app feature still works the same?
I think this was done to avoid making duplicate requests